### ISSUE TITLE
MGHPCC: Replace Brocade switch in NEU cages at 15-43

### ISSUE THREAD

**Issue by naved001** [CCI-MOC/ops-issues#379]
We have tried rebooting and reflashing the firmware, but all switchports stop working after a few hours of reboot.

We can get our replacement switch from engage1 (BU Row 4 Pod A, cage 2 and cage 4)

Here are all the switches in the engage1 brocade fabric

```
RBridge-102# show vcs | include Online
2                10:00:50:EB:1A:A4:29:AC        10.10.10.68     Online           Online               RBridge-2
3                10:00:50:EB:1A:CF:43:28        10.10.10.69     Online           Online               RBridge-3
5                10:00:50:EB:1A:A4:0B:C0        10.10.10.71     Online           Online               RBridge-5
14               10:00:50:EB:1A:AE:7D:04        10.10.10.79     Online           Online               RBridge-14
19               10:00:50:EB:1A:E7:60:CC        10.10.10.83     Online           Online               RBridge-19
101              10:00:50:EB:1A:A4:19:D8        10.10.10.64     Online           Online               RBridge-101
102             >10:00:50:EB:1A:A4:2A:F4*       10.10.10.65     Online           Online               RBridge-102
103              10:00:50:EB:1A:DE:B0:74        10.10.10.66     Online           Online               RBridge-103
104              10:00:50:EB:1A:E7:86:F8        10.10.10.67     Online           Online               RBridge-104
```
Rbirdges lower than 100 are in MIT cages. @msdisme if they have taken those off, then we could use those switches. If not, continue reading ahead.

Rbridge 101 and 104 connect us to CSAIL VLAN 10 which is what we are using for our smaug ocp cluster. so those can't be moved until we connect to CSAIL via a different path.

That leaves rbridge 102 and 103. They are in cage 2 but I don't know the U number. will have to hookup a serial cable and verify the hostname (it should say `RBridge-103#` for 103). We should also verify what's connected to that switch, and if there's any important stuff that breaks if this switch is powered off.

cc: @hakasapl @msdisme @larsks

**Comment by naved001** [comment]
Impact of this switch being down:

1. One openstack controller
2. One oVirt Controller
3. One cache server
4. One Ceph OSD server
5. Five openstack computes
6. one 40G connection to OCT. The other one on 15-42 is currently operational. 
7. ten HIL/BMI nodes

**Comment by pjd-nu** [comment]
If there's anything I can do to accelerate getting this fixed, please let me know. Up to and possibly including driving to Holyoke. (and if necessary picking up a switch on the way - an eBay seller in NH has one with local pickup available)

And I apologize that we (Orran and I) have put you folks in the position of maintaining ancient gear that's starting to fail - we probably should have prioritized doing something about it long ago when they were still running fine.

**Comment by naved001** [comment]
I emailed Morgan from techsquare who informed me that they have some brocades already pulled out of the MIT cages. @hakasapl will replace the switch on Tuesday then.

**Comment by naved001** [comment]
@hakasapl I am documenting how the switchports are connected. 

Currently there's a serial cable between the node at neu-15-12 and the problem brocade-15-43. If you connect that serial cable to the replacement switch then I can start flashing the firmware while the rest of the connections are made.

Port 43 of the IPMI (1G) switch at 15-44 connects to the management interface of brocade-15-43. 

And here's how all ports of brocade-15-43 are connected. TL;DR is ports 1 to 40 connect to nodes between U21 to U40 providing dual 10G connections per node. Ports 41 to 50 are Inter switch links to other brocades. 51 is to a server neu-15-41, and 52 is to OCT/UMass.

Port on brocade-15-43 | Destination | Notes
-- | -- | --
1 | neu-15-21 port1 | dell node
2 | neu-15-21 port2 | dell node
3 | neu-15-22 port1 | dell node
4 | neu-15-22 port2 | dell node
5 | neu-15-23 port1 | dell node
6 | neu-15-23 port2 | dell node
7 | neu-15-24 port1 | dell 730xd (ceph osd node)
8 | neu-15-24 port2 | dell 730xd (ceph osd node)
9 | unknown | please verify before unplugging
10 | unknown (empty ?) | please verify before unplugging
11 | neu-15-26 port1 | cisco node
12 | neu-15-26 port2 | cisco node
13 | neu-15-27 port1 | cisco node
14 | neu-15-27 port2 | cisco node
15 | neu-15-28 port1 | cisco node
16 | neu-15-28 port2 | cisco node
17 | neu-15-29 port1 | cisco node (curator project)
18 | neu-15-29 port2 | cisco node (curator project)
19 | neu-15-30 port1 | cisco node
20 | neu-15-30 port2 | cisco node
21 | neu-15-31 port1 | cisco node
22 | neu-15-31 port2 | cisco node
23 | neu-15-32 port1 | cisco node
24 | neu-15-32 port2 | cisco node
25 | neu-15-33 port1 | cisco node
26 | neu-15-33 port2 | cisco node
27 | neu-15-34 port1 | compute node
28 | neu-15-34 port2 | compute node
29 | neu-15-35 port1 | compute node
30 | neu-15-35 port2 | compute node
31 | neu-15-36 port1 | compute node
32 | neu-15-36 port2 | compute node
33 | neu-15-37 port1 | compute node
34 | neu-15-37 port2 | compute node
35 | neu-15-38 port1 | compute node
36 | neu-15-38 port2 | compute node
37 | neu-15-39 port1 | controller node
38 | neu-15-39 port2 | controller node
39 | neu-15-40 port1 | ovirt1
40 | neu-15-40 port2 | ovirt1
41 | port 41 on brocade-5-43 | Inter Switch Link to rbr6
42 | port 42 on brocade-5-43 | Inter Switch Link to rbr6
43 | port 43 on brocade-5-43 | Inter Switch Link to rbr6
44 | port 44 on brocade-5-43 | Inter Switch Link to rbr6
45 | port 45 on brocade-5-42 | Inter Switch Link to rbr5
46 | port 46 on brocade-5-42 | Inter Switch Link to rbr5
47 | port 47 on brocade-5-42 | Inter Switch Link to rbr5
48 | port 48 on brocade-5-42 | Inter Switch Link to rbr5
49 (FortyGig) | port 49 on brocade-15-42 | Inter Switch Link to rbr1
50 (FortyGig) | port 50 on brocade-3-42 | Inter Switch Link to rbr3
51 (FortyGig) | 40G port on neu-15-41 | supermicro server
52 (FortyGig) | To UMass/OCT switch |

**Comment by naved001** [comment]
This is resolved now; this is what we did:

1. Replaced the physical switch and hooked up a serial cable to the replacement switch
2. Installed NOS 7.0.2  on the switch (all switches in the brocade fabric must be running the same firmware).
3. Once I was able to login to the newly installed switch (still over serial connection), note down the `WWN` of the new switch by running `show vcs`.
4. On the principle node of the brocade fabric, run `vcs replace rbridge-id <node id>`, and then the switch will ask for the WWN on the replacement switch.
5. Then on the replacement switch, `vcs vcsid <cluster id> rbridge-id <node id> logical-chassis enable`, if it fails for some reason, then run `copy default-config startup-config` and reload the switch, and on the next boot it will start joining the fabric.
6. Once the switch joins the fabric, go into bitwarden and apply the 10G and 40G licenses to get all the ports working. 
7. After applying the license, manually bring up those ports (`no shut`). 
8. No further configuration of the switch is required, the new switch will get all configuration from the fabric including all switchport configuration, management IP and management password.
